### PR TITLE
[9.18] Update parameters for repository/commits api

### PR DESCRIPTION
**Pull request by glensc**
Updates Repositories::commits() to match current api documentation:
- https://docs.gitlab.com/ee/api/commits.html#list-repository-commits

```diff
-all
+all: bool
-with_stats
+with_stats: bool
+first_parent: bool
+order: ['default', 'topo']
```

### CODE REVIEW

**Comment by glensc** [comment]
Have you thought to build some bridge code to ease the typing of boolean and date parameters?

looks really tedious to copy `$datetimeNormalizer` and `$booleanNormalizer` around

**Comment by glensc** [comment]
perhaps duplicate: https://github.com/m4tthumphrey/php-gitlab-api/pull/498/files

but this one is more complete.

**Comment by glensc** [comment]
another possibly related PR: https://github.com/m4tthumphrey/php-gitlab-api/pull/476

**Comment by GrahamCampbell** [comment]
Thanks for looking into this. The tests seem to be failing. Could you take a look please @glensc?

File: test/Gitlab/Tests/Api/RepositoriesTest.php [hunk 1]
```diff
@@ -276,11 +276,11 @@ public function shouldGetCommitsWithParams()
         $api = $this->getApiMock();
         $api->expects($this->once())
             ->method('get')
-            ->with('projects/1/repository/commits', array('page' => 2, 'per_page' => 25, 'ref_name' => 'master', 'all' => true, 'with_stats' => true, 'path' => 'file_path/file_name'))
+            ->with('projects/1/repository/commits', array('page' => 2, 'per_page' => 25, 'ref_name' => 'master', 'all' => 'true', 'with_stats' => 'true', 'path' => 'file_path/file_name'))
             ->will($this->returnValue($expectedArray))
         ;
 
-        $this->assertEquals($expectedArray, $api->commits(1, ['page' => 2, 'per_page' => 25, 'ref_name' => 'master', 'all' => true, 'with_stats' => true, 'path' => 'file_path/file_name']));
+        $this->assertEquals($expectedArray, $api->commits(1, ['page' => 2, 'per_page' => 25, 'ref_name' => 'master', 'all' => 'true', 'with_stats' => 'true', 'path' => 'file_path/file_name']));
```
Comment (GrahamCampbell) [inline test/Gitlab/Tests/Api/RepositoriesTest.php]: Why are these changed to strings?

Comment (glensc) [reply test/Gitlab/Tests/Api/RepositoriesTest.php]: I think it's related to how booleans are and should be converted:
- https://github.com/GitLabPHP/Client/issues/519
